pytest/lib/peer.py

```diff
@@ -29,17 +30,28 @@ async def send_raw(self, raw_message):
         self.writer.write(raw_message)
         await self.writer.drain()
 
+    # returns None on timeout
     async def recv(self, expected=None):
         while True:
             response_raw = await self.recv_raw()
+            if response_raw is None:
+                return None
+
             response = BinarySerializer(schema).deserialize(
                 response_raw, PeerMessage)
 
             if expected is None or response.enum == expected:
                 return response
```

Comment:
Change `expected` (enum name) to `expected_fn` a function that determines if this is the message we are waiting for.

```suggestion
    async def recv(self, expected_fn=None):
        while True:
            response_raw = await self.recv_raw()
            if response_raw is None:
                return None

            response = BinarySerializer(schema).deserialize(
                response_raw, PeerMessage)

            if expected_fn is None or expected_fn(response):
                return response
```

Reply:
I don't want to break existing users, so made it both compare and try to call.